tests/unit_tests/test_env_mesh.py

```diff
@@ -31,7 +29,10 @@ def test_load_from_json (self):
    def test_update_agg_cellbox(self):
       self.loaded_env_mesh.update_cellbox(0 , {"x":"5"})
       self.assertEqual (self.loaded_env_mesh.agg_cellboxes[0].get_agg_data()["x"] , "5")
-   
+
```

Comment:
The test 'test_to_tif' fails when run using pytest - 

`=============================================== short test summary info ===============================================
FAILED test_env_mesh.py::TestEnvMesh::test_to_tif - UnboundLocalError: local variable 'data' referenced before assignment
======================================= 1 failed, 2 passed in 291.43s (0:04:51) =======================================
`

Reply:
I get the same error when trying to run from a python console.

Reply:
it was passing before the final refactor to make sonar happier :) ... should pass now

Reply:
@gecoombs can you please pull and try again